frontend/src/scenes/funnels/funnelLogic.ts

```diff
@@ -202,7 +206,11 @@ export const funnelLogic = kea<funnelLogicType>({
             actions.setStepsWithCountLoading(false)
         },
         setFilters: ({ refresh }) => {
-            if (refresh) {
+            // FUNNEL_BAR_VIZ removes the Calculate button
+            // Query performance is suboptimal on psql
+            const autoCalculate =
+                values.featureFlags[FEATURE_FLAGS.FUNNEL_BAR_VIZ] && values.preflight?.is_clickhouse_enabled
```

Comment:
This `autoCalculate` could be a selector. This way the logic doesn't need to be duplicated in two additional components.